Graphics/Vulkan/VulkanGraphicsContext.cpp

```diff
@@ -0,0 +1,397 @@
+        }
+    }
+
+    void VulkanGraphicsContext::DisposeVulkanCommandPool(VkCommandPool vulkanCommandPool) noexcept
+    {
+        if (vulkanCommandPool != VK_NULL_HANDLE)
+        {
+            vkDestroyCommandPool(GetDevice()->GetVulkanDevice(), vulkanCommandPool, nullptr);
+        }
+    }
+
+    void VulkanGraphicsContext::DisposeVulkanFramebuffer(VkFramebuffer vulkanFramebuffer) noexcept
+    {
+        if (vulkanFramebuffer != VK_NULL_HANDLE)
+        {
+            vkDestroyFramebuffer(GetDevice()->GetVulkanDevice(), vulkanFramebuffer, nullptr);
+        }
+    }
+
+    void VulkanGraphicsContext::DisposeVulkanSwapChainImageView(VkImageView vulkanSwapChainImageView) noexcept
+    {
+        if (vulkanSwapChainImageView != VK_NULL_HANDLE)
+        {
+            vkDestroyImageView(GetDevice()->GetVulkanDevice(), vulkanSwapChainImageView, nullptr);
+        }
+    }
+
+    // void VulkanGraphicsContext::BeginCopy(VkImage vulkanImage) noexcept
+    //{
+    // VkImageSubresourceRange subresourceRange{};
+    // subresourceRange.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+    // subresourceRange.levelCount = 1;
+    // subresourceRange.layerCount = 1;
+
+    // VkImageMemoryBarrier vulkanImageMemoryBarrier{};
+    // vulkanImageMemoryBarrier.sType = VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER;
+    // vulkanImageMemoryBarrier.dstAccessMask = VK_ACCESS_TRANSFER_WRITE_BIT;
+    // vulkanImageMemoryBarrier.oldLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+    // vulkanImageMemoryBarrier.newLayout = VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL;
+    // vulkanImageMemoryBarrier.subresourceRange = subresourceRange;
+    // vulkanImageMemoryBarrier.image = vulkanImage;
+
+    // vkCmdPipelineBarrier(GetVulkanCommandBuffer(), VK_PIPELINE_STAGE_HOST_BIT, VK_PIPELINE_STAGE_TRANSFER_BIT, 0, 0,
+    // nullptr, 0, nullptr, 1, &vulkanImageMemoryBarrier);
+    //}
+
+    // void VulkanGraphicsContext::EndCopy(VkImage vulkanImage) noexcept
+    //{
+    // VkImageSubresourceRange subresourceRange{};
+    // subresourceRange.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+    // subresourceRange.levelCount = 1;
+    // subresourceRange.layerCount = 1;
+
+    // VkImageMemoryBarrier vulkanImageMemoryBarrier{};
+    // vulkanImageMemoryBarrier.sType = VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER;
+    // vulkanImageMemoryBarrier.dstAccessMask = VK_ACCESS_TRANSFER_WRITE_BIT;
+    // vulkanImageMemoryBarrier.dstAccessMask = VK_ACCESS_SHADER_READ_BIT;
+    // vulkanImageMemoryBarrier.oldLayout = VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL;
+    // vulkanImageMemoryBarrier.newLayout = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
+    // vulkanImageMemoryBarrier.subresourceRange = subresourceRange;
+    // vulkanImageMemoryBarrier.image = vulkanImage;
+
+    // vkCmdPipelineBarrier(GetVulkanCommandBuffer(), VK_PIPELINE_STAGE_TRANSFER_BIT,
+    // VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT, 0, 0, nullptr, 0, nullptr, 1,
+    //&vulkanImageMemoryBarrier);
+    //}
+
```

Comment:
Whaaaaaaatcha doooooiiiiin hereeeeeee?

Reply:
This is the old API. I kept it around for reference while working on the new cmdlist API. We can most likely delete this now.